src/core/packages/feature-flags/server-internal/src/feature_flags_service.test.ts

```diff
@@ -273,7 +273,7 @@ describe('FeatureFlagsService Server', () => {
     expect(getOverrides()).toStrictEqual({
       'my-overridden-flag': true,
       'myPlugin.myOverriddenFlag': true,
-      myDestructuredObjPlugin: { myOverriddenFlag: true },
+      'myDestructuredObjPlugin.myOverriddenFlag': true,
```

Comment:
we're now flattening the object as we receive the overrides to make sure that we capture all flag names